examples/subplots_axes_and_figures/demo_tight_layout.py

```diff
@@ -40,61 +40,53 @@ def example_plot(ax):
 example_plot(ax2)
 example_plot(ax3)
 example_plot(ax4)
-plt.tight_layout()
+fig.tight_layout()
 
 ###############################################################################
 
 fig, (ax1, ax2) = plt.subplots(nrows=2, ncols=1)
 example_plot(ax1)
 example_plot(ax2)
-plt.tight_layout()
+fig.tight_layout()
 
 ###############################################################################
 
 fig, (ax1, ax2) = plt.subplots(nrows=1, ncols=2)
 example_plot(ax1)
 example_plot(ax2)
-plt.tight_layout()
+fig.tight_layout()
 
 ###############################################################################
 
 fig, axs = plt.subplots(nrows=3, ncols=3)
 for ax in axs.flat:
     example_plot(ax)
-plt.tight_layout()
+fig.tight_layout()
 
 ###############################################################################
 
-fig = plt.figure()
-
+plt.figure()
 ax1 = plt.subplot(221)
 ax2 = plt.subplot(223)
 ax3 = plt.subplot(122)
-
 example_plot(ax1)
 example_plot(ax2)
 example_plot(ax3)
-
 plt.tight_layout()
```

Comment:
I don't understand why you didn't change this (and all the rest) to `fig.tight_layout` instead of removing `fig`?

Reply:
Because the whole subexample is pyplot-based, and I didn't want to change that?

Reply:
Ah, sure.